src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -190,23 +180,60 @@ public static Type parseType(String type) throws ParseException {
         return new Type(trimmedType);
     }
 
+    /**
+     * Parses a {@code String datetime} into an {@code LocalDateTime}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code dateTimeString} is invalid.
+     */
+    public static DateTime parseDateTime(String dateTimeString) throws ParseException {
+        requireNonNull(dateTimeString);
+        String trimmedDateTime = dateTimeString.trim();
+        if (trimmedDateTime.isBlank()) {
+            return new DateTime(trimmedDateTime);
+        }
+
+        if (!DateTime.isValidDateTime(trimmedDateTime)) {
+            throw new ParseException(DateTime.MESSAGE_CONSTRAINTS);
+        }
+        return new DateTime(trimmedDateTime);
+    }
+
     /**
      * Parses a {@code String category} into a {@code Category}.
      * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code category} is invalid.
      */
-    public static Category parseCategory(String category) {
+    public static Category parseCategory(String category) throws ParseException {
         requireNonNull(category);
         String trimmedCategory = category.trim();
+        if (trimmedCategory.isBlank()) {
+            return new Category(trimmedCategory);
```

Comment:
Does this just make the category blank? Do we want to instead, give a default category like "Others" or "Uncategorized" instead? 😃

Reply:
Not really, this part of the code is slightly unintuitive, i have refactored it in my latest change. 
Summary: The handling of empty values is dealt with inside the DateTime class itself. So the Parser class is solely responsible for parsing the arguments. How this argument is dealt it by the DateTime class